File: 0383-ransom-note/0383-ransom-note.cpp
class Solution {
public:
    bool canConstruct(string ransomNote, string magazine) {
        vector<int> v(26);
        for(char ch : ransomNote) {
            v[ch - 'a'] += 1;
        }
        for(char ch : magazine) {
            v[ch - 'a'] -= 1;
        }
        for(char ch : ransomNote) {
            if(v[ch-'a'] > 0) return false;
        }
        return true;

    }
};

File: 0383-ransom-note/0383-ransom-note.java
class Solution {
    public boolean canConstruct(String a, String b) {
        int arr1[] = new int[26];
        int arr2[] = new int[26];
        for(char ch : b.toCharArray()) arr1[ch - 'a']++;
        for(char ch : a.toCharArray()) arr2[ch -'a']++;
        for(char ch : a.toCharArray()) {
            if(arr1[ch - 'a'] < arr2[ch - 'a']) return false;
        }
        return true;
    }
}